### ISSUE TITLE
@service decorator fails on older browsers like IE11, Chrome 41 (Googlebot)

### ISSUE THREAD

**Issue by raido** [ember-decorators/ember-decorators#426]
After installing `ember-cli-babel-polyfills` - to compensate for missing `Symbol` features in IE11, Chrome 41. I hit a error below:

```
An element descriptor's .kind property must be either "method" or "field", but a decorator created an element descriptor with .kind "undefined"
```

And it is caused by `@service` decorator if you don't pass explicit service name to it.

Classic component like this:

```
export default Component.extend({
    myService: service();
});
```

Converted to TS class like this:

```
export default class MyComponent extends Component {
  @service myService !:MyServiceService;
});
```

Will cause it to throw above exception. If I change it to this, with explicit service name passed to decorator:

```
export default class MyComponent extends Component {
  @service('my-service') myService !:MyServiceService;
});
```

It works and no longer throws an exception. 

Although I think it should also work without explicit service name. At least it works on ever green browsers without explicit `serviceName?` but not with older browsers.

**Comment by buschtoens** [comment]
What version of `ember-decorators` do you _think_ are using? v6? If so, you are experiencing this error: #445

**Comment by justinross** [comment]
I'm experiencing this issue, as well, but did not install ember-cli-babel-polyfills. (I'm also not converting to TS, but providing a name to @service resolves the issue. 

`@service('session') session;` works, while
`@service session;` does not.

I'm not using @layout, so I'm not seeing error #445 

I'm using ember-decorators 5.1.4

**Comment by rwjblue** [comment]
Hmm, is this the same issue as https://github.com/emberjs/ember.js/issues/18075? I'm wondering because it seems specifically to affect IE11 when using a Symbol polyfill, which seems very much like what we are working through over in that Ember issue...

@buschtoens - What do you think?

**Comment by justinross** [comment]
If it helps at all, I’m seeing this issue in (and targeting) Firefox ESR 45. Firefox 60 works fine. Haven’t had a chance to test others in between.

**Comment by buschtoens** [comment]
@justinross And you are definitely not using any polyfill? Can you also please install `ember-cli-dependency-lint` to rule out any accidental dependency clobbering by transitive dependencies?

**Comment by justinross** [comment]
@buschtoens further info:
I haven't installed any polyfills. I'll install `ember-cli-dependency-lint` shortly, but I thought this might be useful, first: 

I managed to narrow things down a bit. I'm only getting this error with two of my `@service` uses. One is calling `@service cookies;` (from the `ember-cookies` addon) in my Application adapter, but making the same `@service cookies;` declaration in an `ember-simple-auth` authenticator does not throw the error. 

The other problematic `@service` usage is calling `@service router;` in one of my controllers (oddly enough, others work fine with that syntax). 

![image](https://user-images.githubusercontent.com/1120106/59125205-824ed100-8916-11e9-8801-b5a0df38d0c4.png)

vs

![image](https://user-images.githubusercontent.com/1120106/59125230-92ff4700-8916-11e9-9f98-8000d2bd534b.png)

That seems pretty wonky. I'll go test with `ember-cli-dependency-lint` right now.

**Comment by justinross** [comment]
Okay, installed and ran dependency-lint.

This was the only conflict found: 
![image](https://user-images.githubusercontent.com/1120106/59125925-8e3b9280-8918-11e9-8535-f3ecf978fba4.png)

Since it's not crucial to my app, I just removed `ember-cli-new-version`. No dependency conflicts following its removal. No change in the reported issue, however.

Also, follow-up to my last comment. I *thought* that I had successfully changed most of my `@service` calls back to `@service serviceName`, but I just noticed that using `@service session;` instead of @service('session') session;` in a route causes this error when trying to create a link to that route in a template:

`Error: Assertion Failed: You attempted to define a `{{link-to "admin"}}` but did not pass the parameters required for generating its dynamic segments. An element descriptor's .kind property must be either "method" or "field", but a decorator created an element descriptor with .kind "undefined"`
